File: server-nodejs/models/GoldPrice.js
const { DataTypes } = require('sequelize');

module.exports = (sequelize) => {
  const GoldPrice = sequelize.define('GoldPrice', {
    id: {
      type: DataTypes.INTEGER,
      primaryKey: true,
      autoIncrement: true
    },
    type: {
      type: DataTypes.STRING,
      allowNull: false,
      unique: true
    },
    name: {
      type: DataTypes.STRING,
      allowNull: false
    },
    karat: {
      type: DataTypes.STRING,
      allowNull: false
    },
    purity: {
      type: DataTypes.STRING,
      allowNull: false
    },
    buy_price: {
      type: DataTypes.INTEGER,
      allowNull: false
    },
    sell_price: {
      type: DataTypes.INTEGER,
      allowNull: true
    },
    updated_at: {
      type: DataTypes.STRING,
      allowNull: false
    }
  }, {
    tableName: 'gold_prices',
    timestamps: false
  });

  return GoldPrice;
}; 

File: server-golang/internal/models/goldprice.go
package models

import "time"

// GoldPrice represents a gold price record
type GoldPrice struct {
	ID        string    `json:"id" gorm:"primaryKey"`
	Type      string    `json:"type" gorm:"uniqueIndex;not null"`
	Name      string    `json:"name" gorm:"not null"`
	Karat     string    `json:"karat"`
	Purity    string    `json:"purity"`
	BuyPrice  float64   `json:"buy_price" gorm:"not null"`
	SellPrice float64   `json:"sell_price" gorm:"not null"`
	Company   string    `json:"company"`
	UpdatedAt time.Time `json:"updated_at" gorm:"index;not null"`
	CreatedAt time.Time `json:"created_at" gorm:"autoCreateTime;not null"`
}
